docs/index.adoc

```diff
@@ -13,6 +13,135 @@ This C++ baremetal concurrency library provides a low-level API for operations
 that must be protected in a concurrent environment, without having to know
 platform details.
 
+=== Synopsis
+
+The following headers are available:
+
+* https://github.com/intel/cpp-baremetal-concurrency/blob/main/include/conc/atomic.hpp[`atomic.hpp`]
+* https://github.com/intel/cpp-baremetal-concurrency/blob/main/include/conc/concurrency.hpp[`concurrency.hpp`]
+
+== `atomic.hpp`
+
+`atomic.hpp` contains function templates in the `atomic` namespace. These
+functions are low-level atomic operations that can be used to implement the
+corresponding member functions on
+https://en.cppreference.com/w/cpp/atomic/atomic[`std::atomic`].
+
+[source,cpp]
+----
+template <typename T>
+[[nodiscard]] auto load(T const &t,
+                        std::memory_order mo = std::memory_order_seq_cst) -> T;
+
+template <typename T, typename U>
+auto store(T &t, U value,
+           std::memory_order mo = std::memory_order_seq_cst) -> void;
+
+template <typename T, typename U>
+[[nodiscard]] auto exchange(
+    T &t, U value, std::memory_order mo = std::memory_order_seq_cst) -> T;
+
+template <typename T, typename U>
+auto fetch_add(T &t, U value,
+               std::memory_order mo = std::memory_order_seq_cst) -> T;
+
+template <typename T, typename U>
+auto fetch_sub(T &t, U value,
+               std::memory_order mo = std::memory_order_seq_cst) -> T;
+
+template <typename T, typename U>
+auto fetch_and(T &t, U value,
+               std::memory_order mo = std::memory_order_seq_cst) -> T;
+
+template <typename T, typename U>
+auto fetch_or(T &t, U value,
+              std::memory_order mo = std::memory_order_seq_cst) -> T;
+
+template <typename T, typename U>
+auto fetch_xor(T &t, U value,
+               std::memory_order mo = std::memory_order_seq_cst) -> T;
+----
+
+=== Customization of atomic operations
+
+By default, the atomic interface surfaced in the `atomic` namespace is
+implemented by `atomic::standard_policy`, a policy class that has static
+functions corresponding to each of the above functions. And the
+`standard_policy` functions call
+https://llvm.org/docs/Atomics.html#libcalls-atomic[the compiler intrinsics]
```

Comment:
question: are these intrinsics the same as gcc's?

Reply:
(I'm assuming "yes", if so, I think the documentation should point that out)

Reply:
Yes. Added a note.